### ISSUE TITLE
invalid loop-invariant-statement detected

### ISSUE THREAD

**Issue by tauferp** [tonybaloney/perflint#39]
Following sample yields incorrect loop-invariant-statement

```
def test():
    dicts = ({'1': 'a', '2': 'b'}, {'3': 'c'})
    for d in dicts:
        print({int(k): v for k, v in d.items()})

```

`sample.py:4:15: W8201: Consider moving this expression outside of the loop. (loop-invariant-statement)`

**Comment by jenstroeger** [comment]
And another case:
```python
def test():
    d = {'key': 'some value here'}
    o = object()

    for key, value in d.items():
        if key == 'foo':
            s1, *_ = value.split(' ')
            a = getattr(o, s1)  # W8201: Consider moving this expression outside of the loop. (loop-invariant-statement)
            return a
```
It looks like `a = getattr(o, s1)` is independent of the loop variables `key` and `value`, but `s1` ← `value.split(' ')` depends on a loop variable and therefore isn’t invariant.
